shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/TxData.hs

```diff
@@ -768,8 +768,8 @@ instance
   where
   fromCBOR = decodeRecordNamed "TxOut" (const 2) $ do
     addr <- fromCBOR
-    (b :: Word64) <- fromCBOR
-    pure $ TxOut addr (Coin $ toInteger b)
+    coin <- fromCBOR
+    pure $ TxOutCompact addr coin
```

Comment:
Is this bypassing some important validation? If so, and it's crucial, then we should reconsider.

Reply:
decoding as a Word64 is good, that is consistent with how `Coin` is decoded, but we want to still be able to encode `Coin` values that are out of range, ie as an Integer.

Reply:
as @mrBliss said in the other thread, this can't happen since the coin in the output is stored in memory as a Word64.